### DISCUSSION TITLE
Global `process=false` flag

### DISCUSSION THREAD

**Discussion by jmduke** [markdoc/markdoc#504] [Ideas]
We're porting our docs to Markdoc (backed by Keystatic) and got bitten pretty hard by the issue described in https://github.com/markdoc/markdoc/issues/72 and https://github.com/markdoc/markdoc/issues/306 because our docs have a lot of handlebar-ish code blocks, and it's somewhat hard to enforce the `process=false` tag in every single code block. @rpaul-stripe mentioned adding a global configuration for this and I'm opening this issue to formally suggest that.

(I would also argue that the default should probably be `process=false` rather than `process=true`, but that's a different discussion!)

**Comment by mfix-stripe** [comment]
@jmduke you can handle this by creating a custom [`fence` Node](https://markdoc.dev/docs/nodes) and setting the new default. 

Pseudocode:
```js
import {nodes} from '@markdoc/markdoc'

// ...
nodes: {
  fence: {
    ...nodes.fence,
    attributes: {
      ...nodes.fence.attributes,
      process: { ...nodes.fence.attributes.process, default: false },
    }
  }
}

**Reply by archiewood** [reply]
Thank you for your suggestion

I'm trying to render markdoc code inside code fences, using markdoc (but not process and convert them into tags)

I have been somewhat successful in preventing the tags being processed, but they now tags not displayed in the rendered fences.

Is there a way to retain and render the tags?

## Sample code


```ts
import Markdoc from '@markdoc/markdoc';

export function transformVanillaMarkdown(markdown: string) {
  const vanillaConfig = {
    nodes: {
      fence: {
        ...Markdoc.nodes.fence,
        attributes: {
          ...Markdoc.nodes.fence.attributes,
        process: {
          ...Markdoc.nodes.fence.attributes.process,
          default: false
        }
      }
    }
  }

  const ast = Markdoc.parse(markdown);
  const renderableTree = Markdoc.transform(ast, vanillaConfig);
  return Markdoc.renderers.html(renderableTree);
}
```

## Input markdoc

````markdown
```markdown
# Welcome

Try writing some markdown here...

{% callout type="info" title="Markdown Tips" %}
You can use regular markdown syntax like **bold** and *italic*.
{% /callout %}
```
````
## Result

```html
<pre data-language="markdown"># Welcome

Try writing some markdown here...

You can use regular markdown syntax like **bold** and *italic*.
</pre>
```

## Expected Result

```html
<pre data-language="markdown"># Welcome

Try writing some markdown here...

{% callout type="info" title="Markdown Tips" %}
You can use regular markdown syntax like **bold** and *italic*.
{% /callout %}
</pre>
```
